estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,89 @@
+from datetime import date
+
+from odoo import fields, models, api
+from odoo.tools import date_utils
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Offers made on a listing"
+    _sql_constraints = [
+        (
+            "check_price",
+            "CHECK(price > 0)",
+            "Price of an offer should be only positive",
+        ),
+    ]
+    _order = "price desc"
+
+    price = fields.Float()
+    status = fields.Selection(
+        string="Status",
+        selection=[
+            ("accepted", "Accepted"),
+            ("refused", "Refused"),
+        ],
+        copy=False,
+    )
+    partner_id = fields.Many2one("res.partner", string="Buyer", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+    property_type_id = fields.Many2one(
+        related="property_id.property_type_id", store=True
+    )
+    validity = fields.Integer(default=7, string="Validity (days)")
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline",
+        inverse="_inverse_date_deadline",
+        string="Deadline",
+    )
+
+    @api.depends("validity")
+    def _compute_date_deadline(self):
+        for offer in self:
+            create_date_actual = (
+                date.today() if not offer.create_date else offer.create_date.date()
+            )
+            offer.date_deadline = date_utils.add(
+                create_date_actual, days=offer.validity
+            )
+
+    def _inverse_date_deadline(self):
+        for offer in self:
+            create_date_actual = (
+                date.today() if not offer.create_date else offer.create_date.date()
+            )
+            offer.validity = (offer.date_deadline - create_date_actual).days
+
+    def action_offer_accept(self):
+        for offer in self:
+            offer.status = "accepted"
+            offer._action_check_offers()
+        return True
+
+    @api.depends("status")
+    def _action_check_offers(self):
+        if self.status == "accepted":
+            if self.property_id.state in ("offer-accepted", "sold"):
+                self.status = False
+                raise UserError("An offer has already been accepted!")
+            else:
+                self.property_id.state = "offer-accepted"
+                self.property_id.selling_price = self.price
+                self.property_id.buyer_id = self.partner_id
+        return True
+
+    @api.depends("status")
+    def action_offer_refuse(self):
+        for offer in self:
+            offer.status = "refused"
```

Comment:
tip: this is equivalent
```suggestion
        self.status = "refused"
```